File: Lista 1/20/20.py
"""
Escrever um programa declarando três variáveis do tipo inteiro (a, b e c). Ler um valor maior que zero para cada 
variável  (se  o  valor  digitado  não  é  válido,  mostrar  mensagem  e  ler  novamente).  Exibe  o  menor  valor  lido 
multiplicado pelo maior e o maior valor dividido pelo menor.
"""

a = 0
b = 0
c = 0

bigger = 0 
smaller = 0

temp = 0 

while True:
    temp = int(input("A value: "))
    if temp <= 0:
        print("invalid number! try again...")
        continue
    else:
        a = temp
    temp = int(input("B value: "))
    if temp <= 0:
        print("invalid number! try again...")
        continue
    else:
        b = temp
    temp = int(input("C value: "))
    if temp <= 0:
        print("invalid number! try again...")
        continue
    else:
        c = temp
    break;

if a > b and a > c:
    bigger = a
    if b < c:
        smaller = b
    else:
        smaller = c

if b > a and b > c:
    bigger = b
    if a < c:
        smaller = a
    else:
        smaller = c

if c > a and c > b:
    bigger = c
    if a < b:
        smaller = a
    else:
        smaller = b 

print(f"Smaller number {smaller} * bigger number {bigger} = {smaller * bigger}")
print(f"Bigger number {bigger} / smaller number {smaller} = {bigger/smaller}")

File: List 1 - finished/20.cs
/*
* Escrever um programa declarando três variáveis do tipo inteiro (a, b e c). Ler um valor maior que zero para cada 
* variável  (se  o  valor  digitado  não  é  válido,  mostrar  mensagem  e  ler  novamente).  Exibe  o  menor  valor  lido 
* multiplicado pelo maior e o maior valor dividido pelo menor.
*/

int a,b,c;

a = ReturnValue("A");
b = ReturnValue("B");
c = ReturnValue("C");

int bigger = Math.Max(a, Math.Max(b,c));
int smaller = Math.Min(a, Math.Min(b,c));

Console.WriteLine($"Smaller ({smaller}) * Bigger ({bigger}) = {smaller * bigger}");
Console.WriteLine($"Bigger ({bigger}) / Smaller ({smaller}) = {bigger/smaller:f2}");


int ReturnValue(string variableName)
{
    int value;
    while (true)
    {
        Console.WriteLine($"Entry a valid value for variable {variableName}");
        value = int.Parse(Console.ReadLine());
        if(value > 0)
        {
            return value;
        }
        Console.WriteLine("Invalid number...");
    }
}